### PR TITLE
feat: add e2e for llm o11y

### PR DESCRIPTION
**Pull request by tewarig**
## Pull Request

---

### 📄 Summary

This is PR ads basic happy flow E2E for our LLM observability flow. Since most of the cases are covered by the integration test itself, we have added happy path here.

We have also added test for Test tab here. 

So we have added an integration test for a test endpoint here. 

#### Issues closed by this PR
Closes - https://github.com/SigNoz/engineering-pod/issues/5764
---

### ✅ Change Type
_Select all that apply_

- [x] 🧪 Test-only — E2E + integration coverage for LLM Observability
- [x] 🛠️ Infra / Tooling — E2E helpers + e2e-scoped feature-flag conftest

---

### 🧪 Testing Strategy

- **Tests added:**
  - **E2E (Playwright):** `tests/e2e/tests/llm-o11y/attribute-mapping.spec.ts`, `tests/e2e/tests/llm-o11y/llm-pricing.spec.ts` + `tests/e2e/helpers/{attribute-mapping,llm-pricing}.ts`. Both specs are `test.describe.configure({ mode: 'serial' })` and tear down what they create via the API.
  - **Integration (Jest/RTL/MSW):** `TestTab/__tests__/TestTab.test.tsx` — 4 cases:

    | Case | Time |
    |---|---|
    | runs the sample span through the mappers and renders the populated result | 447 ms |
    | surfaces a backend error and renders no results | 122 ms |
    | persists an edited span to local storage and restores it on remount | 657 ms |
    | resets to the sample span and clears the persisted input | 70 ms |

- **Manual verification:** full LLM Observability Jest suite is green locally — **7 suites / 70 tests passed in 24.3 s** (`npx jest --verbose src/container/LLMObservability`), including the 4 new Test-tab cases.
- **Edge cases covered:** Test-tab backend 500 → error surfaced and no results rendered; span input persisted across remount and cleared on reset; E2E fixture cleanup is failure-tolerant (`.catch()` on list/delete so a broken run doesn't mask the real assertion failure).
- **Not yet captured:** a fully-green E2E run in CI — the e2e workflow only fires on PRs carrying the `safe-to-e2e` label (see Notes).

---

### ⚠️ Risk & Impact Assessment

- **Blast radius:** none in product code. No `frontend/src` runtime file, no Go package, and no shared pytest fixture is modified — the only non-test-file change is the new `tests/e2e/conftest.py`, scoped to the `e2e` package.
- **Potential regressions:** limited to CI surface. `tests/e2e/conftest.py` overrides the package-scoped `signoz` fixture with a distinct `cache_key`, so it brings up **one additional container set** for the e2e package rather than mutating the shared one — integration suites keep the stock feature set.
- **Rollback plan:** revert the PR. No schema/migration changes, no runtime behaviour to unwind.

---

### 📝 Changelog

| Field | Value |
|------|-------|
| Deployment Type | N/A |
| Change Type | Maintenance |
| Description | N/A — test-only, no user-facing change. |

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested (LLM Observability Jest suite green: 7 suites / 70 tests; E2E specs exercised against a local flag-enabled stack)
- [x] Breaking changes documented (none)
- [x] Backward compatibility considered

---

## Notes for Reviewers

### CODE REVIEW

File: frontend/src/container/LLMObservability/AttributeMapping/TestTab/__tests__/TestTab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+	it('persists an edited span to local storage and restores it on remount', async () => {
+		const user = userEvent.setup({ pointerEventsCheck: 0 });
+		const { unmount } = render(<LLMObservabilityAttributeMapping />);
+
+		await user.click(screen.getByRole('tab', { name: 'Test' }));
+		await screen.findByTestId('run-test-button');
+
+		expect(screen.getByTestId('monaco')).toHaveValue(SAMPLE_SPAN_JSON);
+
+		fireEvent.change(screen.getByTestId('monaco'), {
```
Comment (aks07) [inline frontend/src/container/LLMObservability/AttributeMapping/TestTab/__tests__/TestTab.test.tsx]: can we remove fireEvents?

File: tests/e2e/conftest.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# TODO: remove this file once ai-o11y goes live
```
Comment (AshwinBhatkal) [inline tests/e2e/conftest.py]: can you add the exact name here - enable_ai_observability?
Easier during cleanup

File: tests/e2e/helpers/llm-pricing.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	await page.getByTestId(LLM_PRICING_PAGE_TESTID).waitFor({ state: 'visible' });
+}
+
+interface PricingRule {
+	id: string;
+	modelName: string;
+}
+
+/** List pricing rules via API (`GET /api/v1/llm_pricing_rules`). */
+export async function listPricingRules(
```
Comment (AshwinBhatkal) [inline tests/e2e/helpers/llm-pricing.ts]: do we need export here?

File: tests/e2e/helpers/attribute-mapping.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+		.waitFor({ state: 'visible' });
+}
+
+interface SpanMapperGroup {
+	id: string;
+	name: string;
+}
+
+/** List span-mapper groups via API (`GET /api/v1/span_mapper_groups`). */
+export async function listSpanMapperGroups(
```
Comment (AshwinBhatkal) [inline tests/e2e/helpers/attribute-mapping.ts]: same here

File: tests/e2e/helpers/llm-o11y/attribute-mapping.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import type { APIRequestContext, Page } from '@playwright/test';
```
Comment (AshwinBhatkal) [inline tests/e2e/helpers/llm-o11y/attribute-mapping.ts:1]: put these 2 helpers in a folder for llmobservability

File: frontend/src/container/LLMObservability/AttributeMapping/TestTab/__tests__/TestTab.test.tsx (hunk 1 above)
Comment (tewarig) [reply frontend/src/container/LLMObservability/AttributeMapping/TestTab/__tests__/TestTab.test.tsx]: done

File: tests/e2e/helpers/llm-pricing.ts (hunk 3 above)
Comment (tewarig) [reply tests/e2e/helpers/llm-pricing.ts]: not needed

File: tests/e2e/helpers/llm-o11y/attribute-mapping.ts (hunk 5 above)
Comment (AshwinBhatkal) [reply tests/e2e/helpers/llm-o11y/attribute-mapping.ts:1]: nit: tests you have it named llm-o11y. I didn't see the short form then. Sorry 😅

Comment (tewarig) [reply tests/e2e/helpers/llm-o11y/attribute-mapping.ts:1]: done

**Review by AshwinBhatkal — APPROVED** [review]
